Deduplicate headers in generated GeneratedSerializers file

szewai opened the pull request:
#### 40ad311960214102f5e5b80de441cfa280598b76
<pre>
Deduplicate headers in generated GeneratedSerializers file
<a href="https://bugs.webkit.org/show_bug.cgi?id=254547">https://bugs.webkit.org/show_bug.cgi?id=254547</a>
rdar://107283996

Reviewed by Alex Christensen.

Use dictionary and set to avoid including a header multiple times in generated file.

* Source/WebKit/Scripts/generate-serializers.py:
(ConditionalHeader.__eq__):
(ConditionalHeader):
(ConditionalHeader.__hash__):
(main):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder&lt;Namespace::ConditionalCommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::ConditionalCommonClass&gt;::decode):
(IPC::ArgumentCoder&lt;Namesapce::CommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namesapce::CommonClass&gt;::decode):
(IPC::ArgumentCoder&lt;Namesapce::AnotherCommonClass&gt;::encode):
(IPC::ArgumentCoder&lt;Namesapce::AnotherCommonClass&gt;::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:

Canonical link: <a href="https://commits.webkit.org/262625@main">https://commits.webkit.org/262625@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/539f888a5eac588f945c69595a340582e75dc67d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2069 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2099 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2165 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2995 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2132 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2180 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2143 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1887 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2088 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1856 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1853 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2845 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/2101 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1841 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1829 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1738 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1907 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1877 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3009 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1903 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1700 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1831 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1840 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2009 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/228 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (changes requested):
Modify Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in and run make -C Source/WebKit/Scripts/webkit/tests to reflect this change in the tests.

szewai committed:
Deduplicate headers in generated GeneratedSerializers file
https://bugs.webkit.org/show_bug.cgi?id=254547
rdar://107283996

Reviewed by Alex Christensen.

Use dictionary and set to avoid including a header multiple times in generated file.

* Source/WebKit/Scripts/generate-serializers.py:
(ConditionalHeader.__eq__):
(ConditionalHeader):
(ConditionalHeader.__hash__):
(main):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp:
(IPC::ArgumentCoder<Namespace::ConditionalCommonClass>::encode):
(IPC::ArgumentCoder<Namespace::ConditionalCommonClass>::decode):
(IPC::ArgumentCoder<Namesapce::CommonClass>::encode):
(IPC::ArgumentCoder<Namesapce::CommonClass>::decode):
(IPC::ArgumentCoder<Namesapce::AnotherCommonClass>::encode):
(IPC::ArgumentCoder<Namesapce::AnotherCommonClass>::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h:
* Source/WebKit/Scripts/webkit/tests/SerializedTypeInfo.cpp:
(WebKit::allSerializedTypes):
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in:

Canonical link: https://commits.webkit.org/262625@main
